src/sentry/search/django/backend.py

```diff
@@ -250,6 +257,80 @@ def query(self, project, tags=None, environment=None, sort_by='date', limit=100,
         else:
             retention_window_start = None
 
+        # TODO: This could also do something janky like switching based on the current
+        #       tagstore backend, or a global option, or a number of other things. Going
+        #       by the project seemed the best for now.
+        use_snuba = bool(project.get_option('sentry:snuba', False))
+        if use_snuba and (tags or
+                          environment is not None or
+                          any(key in parameters for key in ('date_from', 'date_to'))):
+
+            # if we're in this branch then Snuba can help us filter down the eligible group ids
+            filters = {
+                'project_id': [project.id],
+            }
+
+            if environment is not None:
+                filters['environment'] = [environment.id]
+
+            conditions = []
+            for tag, val in six.iteritems(tags):
+                col = 'tags[{}]'.format(tag)
+                if val == ANY:
+                    conditions.append((col, 'IS NOT NULL', None))
```

Comment:
Just realized that the SQL expression that the snuba service uses for tag expansion (`tags.value[indexOf(tags.key, 'foo')]` returns the empty string instead of `NULL` so the correct expression here might be `(col, '!=', "")` 

But the empty string is maybe ambigious so I'll see if theres a way to make it actually Null